### ISSUE TITLE
can't reference a CTE column when doing a join "using"

### ISSUE THREAD

**Issue by ngbusca** [prestodb/presto#15540]
The following query fails in presto 0.217 (used by athena aws) but works fine on 0.172:

`
with db0 as (select 1 as key,2 as value),
db1 as (select 1 as key, 3 as value)
select db0.key from db0 join db1 using(key)
Result: SYNTAX_ERROR: line 3:8: Column 'db0.key' cannot be resolved
`

Doing instead:
`
with db0 as (select 1 as key,2 as value),
db1 as (select 1 as key, 3 as value)
select db0.key from db0 join db1 on db0.key=db1.key
`
works as expected on both 0.217 and 0.172.
Is this the expected behavior of the later version?

**Comment by rongrong** [comment]
It should be 
```
with db0 as (select 1 as key,2 as value), db1 as (select 1 as key, 3 as value) select *key* from db0 join db1 using(key)
```

Please read the details in the "USING" section in https://prestodb.io/docs/current/sql/select.html

**Comment by rongrong** [comment]
The previous behavior is not correct and was fixed in 0.196: https://github.com/prestodb/presto/pull/9938

**Comment by ngbusca** [comment]
makes sense, thanks!
